dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2639,327 @@ void FileSprayer::pushParts()
     performTransfer();
 }
 
+using namespace Azure::Storage;
+using namespace Azure::Storage::Files;
+using namespace Azure::Storage::Blobs;
+
+enum ApiCopyStatus { Pending, Success, Failed, Aborted };
+interface IAPICopyClient : implements IInterface
+{
+    virtual ApiCopyStatus startCopy(const char *target, const char * source) = 0;
+    virtual ApiCopyStatus getProgress(CDateTime & dateTime, int64_t & outputLength) = 0;
+    virtual ApiCopyStatus abortCopy() = 0;
+    virtual ApiCopyStatus getStatus() const = 0;
+};
+
+class CApiCopyClient : implements IAPICopyClient, public CInterface
+{
+protected:
+    ApiCopyStatus status;
+public:
+    IMPLEMENT_IINTERFACE
+    virtual ApiCopyStatus getStatus() const
+    {
+        return status;
+    }
+    static bool buildAzureStorageURIBase(IStoragePlane * storagePlane, StringBuffer & uri)
+    {
+        const char *accountName = storagePlane->queryStorageApiAccount();
+        if (isEmptyString(accountName))
+        {
+            UWARNLOG("Unable to API copy: no account name provided for storage plane");
+            return false;
+        }
+        uri.appendf("https://%s", accountName);
+
+        if (storagePlane->getStorageType()==StorageType::StorageTypeAzureFile)
+        {
+            uri.append(".file");
+        }
+        else
+        {
+            dbgassertex(storagePlane->getStorageType()==StorageType::StorageTypeAzureBlob);
+            uri.append(".blob");
+        }
+        const char *container = storagePlane->queryStorageContainer();
+        if (isEmptyString(container))
+        {
+            UWARNLOG("Unable to API copy: no container name provided for storage plane");
```

Comment:
see above comment re. useAPICopy() and exception

Reply:
I've moved this check to useApiCopy() and removed the test here.  I don't think it's necessary to assertex(!isEmptyString(accountName)) here.